### ISSUE TITLE
JioMart - compliance check

### ISSUE THREAD

**Issue by bluecypher** [ONDC-Official/verification-logs#536]
### Flow 1

**/search**
Buyer app finder fee type should be "percent"
GPS Lat/Long Precision should be same

**/select**
fulfillments GPS Lat/Long Precision should be same

**/init**
items[].fulfillment_id missing
add_ons and offers are invalid keys
billing.address areaCode and ward are invalid keys
billing.address.door is deprecated key
area_code in billing.address does not match with area_code in /select
fulfillments[].id is missing
fulfillments[].provider_id is deprecated key
fulfillments[].customer is invalid object
payment object is not as per the API contract
billing object timestamps are missing

**/confirm**
items[].fulfillment_id is missing
Billing object mismatches in /init and /confirm
fulfillments[0].id is missing in /confirm
fulfillments[0].end.person object is missing
payment.params.transaction_id is missing when status is PAID
payment settlement_details mismatch in /on_init & /confirm
order created and updated timestamps are mandatory in /confirm
Buyer App Finder fee can't change

**/update (refund)**
update API for refund calculation is missing

**_No Flow 2 to validate_**
**_Flow 5 is not as per the test case scenario (out of stock)_**

@tusharmoraye

**Comment by tusharmoraye** [comment]
Hi @bluecypher @BLR-0118 
All feedback implemented here #574

**Comment by bluecypher** [comment]
context.city should be one of the [ONDC city codes](https://docs.google.com/spreadsheets/d/1BtABwe4CXgN_sHIvDH51dJeYzRWdAhedYtzx10apOTs/edit#gid=0)

@tusharmoraye

**Comment by tusharmoraye** [comment]
Hi @bluecypher
added city code support #639

**Comment by BLR-0118** [comment]
1. item.quantity.count used in select & other APIs is integer. However, in the flows, it alternates between integer & float in the different flows. why?

**Comment by BLR-0118** [comment]
@tusharmoraye

**Comment by tusharmoraye** [comment]
Hi @BLR-0118,
The quantity count is of integer type in all flows. While logging, it was logged as a float value. I have fixed this in #682 .

**Comment by BLR-0118** [comment]
Flow 2
====
1. timestamp for /on_init is earlier than /init and has different message_id. It seems to be for the same transaction as the txn id is the same. If this happens, the expected behaviour of buyer app is to NACK the /on_init;
Other issues
========
1. Not clear if unsolicited on_status calls are handled as the logs show status going from accepted to completed. In many of the logs, the pickup time is wrong. What's the behaviour of the buyer app in this case?
2. Is the confirm, on_confirm flow (as documented in the API contract) enabled?
3. Not clear why the gps decimal precision is different for different locations. While it mostly meets the minimum precision requirements (6 decimals), what's the reason for this?

**Comment by BLR-0118** [comment]
@tusharmoraye

**Comment by tusharmoraye** [comment]
Hi @BLR-0118 
1. We submitted another /on_init response due to a manual error. We have submitted the correct /on_init with the correct message id here #701 .
2. Unsolicited calls are in place. The journey at order level post created: Accepted > In Progress > Completed. Since the need for logs didn’t have that mandate to show the entire journey, we had not submitted it.
3. Yes
4. This is because lat - 7 characters, long - 14 characters. We shall trim gps at 6 characters in all requests.
